plugins/network/NetClass.cpp

```diff
@@ -167,21 +109,18 @@ BOOL NetResourceList::Enumerate(
 			if (EnumCode != ERROR_NO_MORE_ITEMS)
 			{
 				Clear();
-				EnumFailed = TRUE;
+				EnumFailed = true;
 			}
 
 			break;
 		}
 
 		if (NetCount > 0)
 		{
-			ResList = (NETRESOURCE*)realloc(ResList, (ResCount + NetCount) * sizeof(*ResList));
-			ZeroMemory(&ResList [ResCount], sizeof(*ResList)*NetCount);
-
-			for (unsigned I = 0; I < NetCount; I++)
-				CopyNetResource(ResList[ResCount + I], nr[I]);
+			ResList.reserve(ResList.size() + NetCount);
```

Comment:
Exact reserve in a loop is usually an antipattern - instead of growing exponentially we're reallocating on every iteration, which kills the amortized constant complexity.

Reply:
I don't have a cycle here
```
if (NetCount > 0)
		{
			ResList.reserve(ResList.size() + NetCount);

			for (size_t i = 0; i < NetCount; i++)
				ResList.emplace_back(NetResource(nr[i]));
		}
```

Reply:
This whole thing is in one big `for(;;)`, isn't it?

Reply:
chang vector to list

Reply:
list is almost always way worse than vector, unless you have specific requirements like no iterators invalidation.

vector is totally fine here, just please remove the explicit `reserve` and let it grow automatically.